### PR TITLE
[Inv-44] - accommodation sorting options and pagination

### PR DESCRIPTION
**Pull request by SurajSulten**
- [x]  Implemented sorting by: 

 - price
 - number of people
 - number of rooms
 - created at 
- [x] implemented pagination

### CODE REVIEW

File: src/modules/accommodations/controllers/accommodations.controller.ts [hunk 1]
```diff
@@ -11,8 +13,23 @@ export class AccommodationsController {
   }
 
   @Get()
-  findAll() {
-    return this.AccommodationService.findAll();
+  @ApiOperation({
+    summary: 'Get list of accommodations that match filtering and sorting',
+  })
+  @ApiResponse({
+    status: 201,
```
Comment (ShahobiddinIsmoilov) [inline src/modules/accommodations/controllers/accommodations.controller.ts:20]: Success status for GET request should not be 201, it should be 200

File: src/modules/accommodations/controllers/accommodations.controller.ts [hunk 2]
```diff
@@ -11,8 +13,23 @@ export class AccommodationsController {
...
+    description: 'Get filtered and sorted accommodations',
+  })
+  @ApiResponse({
+    status: 409,
```
Comment (ShahobiddinIsmoilov) [inline src/modules/accommodations/controllers/accommodations.controller.ts:24]: I think we are not throwing such error anywhere

File: src/shared/transformers/parse-int.transformers.ts
Comment (ShahobiddinIsmoilov) [inline src/shared/transformers/parse-int.transformers.ts:1]: It should be parse-int.transformer.ts, not transformers

File: src/modules/accommodations/controllers/accommodations.controller.ts (hunk 2 above)
Comment (SurajSulten) [reply src/modules/accommodations/controllers/accommodations.controller.ts]: changed to appropriate

File: src/shared/transformers/parse-int.transformers.ts
Comment (SurajSulten) [reply src/shared/transformers/parse-int.transformers.ts:1]: fixed

File: src/modules/accommodations/controllers/accommodations.controller.ts (hunk 1 above)
Comment (SurajSulten) [reply src/modules/accommodations/controllers/accommodations.controller.ts]: fixed

File: src/modules/accommodations/entities/accommodations.entity.ts [hunk 3]
```diff
@@ -67,7 +67,7 @@ export class Accommodation {
   @OneToMany(() => AccommodationImage, (image) => image.accommodation)
   images: AccommodationImage[];
 
-  @OneToOne(() => AccommodationAmenity)
+  @OneToOne(() => AccommodationAmenity, (amenity) => amenity.accommodation)
```
Comment (NegatiV-64) [inline src/modules/accommodations/entities/accommodations.entity.ts:70]: Why was this change required?

File: src/modules/accommodations/services/accommodations.service.ts [hunk 4]
```diff
@@ -1,19 +1,43 @@
...
   ) {}
 
   create() {
     return 'This action adds a new Accommodation';
   }
 
-  findAll() {
-    return `This action returns all Accommodation`;
+  async listAccommodations(searchParams: ListAccommodationsParamsDto) {
+    const page = searchParams.page ?? 1;
```
Comment (NegatiV-64) [inline src/modules/accommodations/services/accommodations.service.ts:20]: Create a util function to actually map the search params to DB query

File: src/modules/accommodations/services/accommodations.service.ts [hunk 5]
```diff
@@ -1,19 +1,43 @@
...
+
+    const sortBy = {
+      price: sortByPrice,
+      numberOfRooms: sortByNumberOfRooms,
+      allowedNumberOfPeople: sortByNumberOfPeople,
+      createdAt: sortByCreatedAt,
+    };
+
+    const accommodations = await this.accommodationRepository.find({
+      skip: (page - 1) * limit,
```
Comment (NegatiV-64) [inline src/modules/accommodations/services/accommodations.service.ts:36]: Create a shared util function for pagination

File: src/modules/auth/types/auth-payload.type.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { Roles } from '#/shared/constants/user-roles.constants';
+import { Roles } from '#/shared/constants/user-roles.constant';
```
Comment (NegatiV-64) [inline src/modules/auth/types/auth-payload.type.ts:1]: Nice catch!

**Review by NegatiV-64 — CHANGES_REQUESTED** [review]

File: src/modules/accommodations/dto/requests/list-accommodations-params.dto.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  @IsOptional()
+  @IsInt()
+  @IsPositive()
+  @Max(1000)
+  limit?: number;
+
+  @IsString()
+  @IsEnum(SortOrder)
+  @IsOptional()
+  sortByPrice?: SortOrder;
```
[the remark this answers is not in the document]
Comment (SurajSulten) [reply src/modules/accommodations/dto/requests/list-accommodations-params.dto.ts:29]: I think in this case we can't sort by two options I mean more than one option. I tried this:
const orderBy = {
      [sortBy]: sortOrder,
};

File: src/modules/accommodations/entities/accommodations.entity.ts (hunk 3 above)
Comment (SurajSulten) [reply src/modules/accommodations/entities/accommodations.entity.ts:70]: it appeared when I pulled updates to main and merged into branch. There was a conflict and I left this. Should I remove it?

File: src/modules/accommodations/dto/requests/list-accommodations-params.dto.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (SurajSulten) [reply src/modules/accommodations/dto/requests/list-accommodations-params.dto.ts:29]: sorry, I think I got your point, will try it

File: src/modules/accommodations/entities/accommodations.entity.ts (hunk 3 above)
Comment (NegatiV-64) [reply src/modules/accommodations/entities/accommodations.entity.ts:70]: Just checked in the docs: https://orkhan.gitbook.io/typeorm/docs/one-to-one-relations

Seems to okay

File: src/modules/accommodations/services/accommodations.service.ts (hunk 4 above)
Comment (SurajSulten) [reply src/modules/accommodations/services/accommodations.service.ts:20]: done

File: database/migrations/1731489197307-fixed-accomodation-amenity-table.ts
Comment (NegatiV-64) [inline database/migrations/1731489197307-fixed-accomodation-amenity-table.ts:1]: I think that we fixed the issue, no?

Comment (SurajSulten) [reply database/migrations/1731489197307-fixed-accomodation-amenity-table.ts:1]: Yes, but it appeared when I ran migration even I pulled last changes. Now I deleted this file and tried to run migration and it wasn't created so there is no this file

**Review by NegatiV-64 — APPROVED** [review]
